pyvrp/read.py

```diff
@@ -198,6 +211,16 @@ def read(
         distances[0, backhaul] = MAX_VALUE
         distances[np.ix_(backhaul, linehaul)] = MAX_VALUE
 
+    if "unit_distance_cost" in instance:
+        # When there are unit distance costs, distances are effectively
+        # scaled/rounded twice: once in the distance matrix, and once in the
+        # unit distance costs. We also need to account for this in the other
+        # objective terms (fixed costs and prizes), so we scale/round them
+        # again. This only works if the fixed cost and prize data does not
+        # have too many decimal places.
+        fixed_costs *= round_func(1)
+        prizes *= round_func(1)
```

Comment:
I don't understand why this isn't covered by the new tests. Perhaps something to figure out and correct with the tests?